doc/source/whatsnew/v1.2.1.rst

```diff
@@ -36,6 +36,76 @@ Fixed regressions
 
 .. ---------------------------------------------------------------------------
 
+.. _whatsnew_121.ufunc_deprecation:
+
+Calling numpy ufuncs on non-aligned DataFrames
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before pandas 1.2.0, calling a numpy ufunc on non-aligned DataFrames (or
+DataFrame / Series combination) would ignore the indices, only match
+the inputs by shape, and use the index/rows of the first DataFrame for
+the result:
+
+.. code-block:: python
+
+    >>> df1 = pd.DataFrame({"a": [1, 2], "b": [3, 4]}, index=[0, 1])
+    ... df2 = pd.DataFrame({"a": [1, 2], "b": [3, 4]}, index=[1, 2])
+    >>> df1
+       a  b
+    0  1  3
+    1  2  4
+    >>> df2
+       a  b
+    1  1  3
+    2  2  4
+
+    >>> np.add(df1, df2)
+       a  b
+    0  2  6
+    1  4  8
+
+This contrasts with how other pandas operations work, which first align
+the inputs:
+
+.. code-block:: python
+
+    >>> df1 + df2
+         a    b
+    0  NaN  NaN
+    1  3.0  7.0
+    2  NaN  NaN
+
+In pandas 1.2.0, we refactored how numpy ufuncs are called on DataFrames,
+and this started to align the inputs first, as happens in other pandas
+operations. For pandas 1.2.1, we restored the previous behaviour to avoid
+a breaking change, but the above example of ``np.add(df1, df2)`` with
+non-aligned inputs will now to raise a warning, and a future pandas 2.0
+release will start aligning the inputs first.
```

Comment:
perhaps add the issue number at the end of this sentance (or somewhere else if you think more appropriate, the release note does not currently have a reference to the issue)